### PR TITLE
internal: Refactor Envoy into v2 package

### PR DESCRIPTION
**Pull request by stevesloka**
Refactors the envoy package into a v2 package for Envoy API V2 types.
Common items are kept in the envoy package for use by other versions.

This doesn't change anything functionally, just refactors envoy into a v2 package. 

Updates: #1898

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

**Comment by stevesloka** [comment]
I'm going to do a 2nd pass to clean up imports and names, etc.

**Comment by skriss** [comment]
@stevesloka and I talked live about this today, but recording some thoughts for others. I'm not thrilled about having to copy-paste all of this stuff (one copy for v2, another likely very similar copy for v3) but I don't have a better idea that will clearly work well at the moment. This'll definitely add to our maintenance burden while we carry both versions, since we'll potentially have to make changes in both, add tests in both.

That said, the specifics of this PR look fine to me if this is the path we're going down. I do think that once we support multiple versions, it may give us some better ideas about ways to make adding new versions in the future easier.

Interested in thoughts from other maintainers here.

**Comment by youngnick** [comment]
Yes, I also don't see a better way to achieve having simultaneous support for both versions for a while, but also agree that if anything is to show us better ways it's trying it out.

Also, the annoyance factor of having to make changes in two places seems like it will make us want to get rid of this situation as soon as we can once we're confident v3 is working.

File: internal/envoy/listener.go [hunk 1]
```diff
@@ -14,27 +14,9 @@
...
-	"time"
 
-	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
-	envoy_api_v2_auth "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
-	envoy_api_v2_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
-	envoy_api_v2_listener "github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
-	accesslog "github.com/envoyproxy/go-control-plane/envoy/config/filter/accesslog/v2"
-	envoy_config_filter_http_ext_authz_v2 "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/ext_authz/v2"
-	lua "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/lua/v2"
 	http "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
```
Comment (youngnick) [inline internal/envoy/listener.go:19]: Does this mean that the HTTP stuff still in here is also xDS v2?

**Review by youngnick — APPROVED** [review]
LGTM, with the already-discussed caveats.

File: internal/envoy/v2/secret.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	// nolint:gosec
```
Comment (skriss) [inline internal/envoy/v2/secret.go:17]: don't think this is needed, see prev comment re: `crypto/sha1`

**Review by skriss — CHANGES_REQUESTED** [review]
Handful of nits, otherwise LGTM

File: internal/envoy/bootstrap.go [hunk 3]
```diff
@@ -10,323 +10,29 @@
...
 	"github.com/golang/protobuf/jsonpb"
 	"github.com/golang/protobuf/proto"
-	"github.com/golang/protobuf/ptypes/any"
-	"github.com/projectcontour/contour/internal/protobuf"
 )
 
-// sdsResourcesSubdirectory stores the subdirectory name where SDS path resources are stored to.
-const sdsResourcesSubdirectory = "sds"
+// SdsResourcesSubdirectory stores the subdirectory name where SDS path resources are stored to.
+const SdsResourcesSubdirectory = "sds"
```
[the remark this answers is not in the document]
Comment (youngnick) [reply internal/envoy/bootstrap.go:27]: In general we should do initialisms capitalised, I think, and regard places where we don't do that as tech debt.

File: internal/envoy/listener.go (hunk 1 above)
Comment (stevesloka) [reply internal/envoy/listener.go:19]: yup good catch, I moved it all into the v2 package.

**Review by skriss — APPROVED** [review]
:shipit:

**Comment by stevesloka** [comment]
I was able to get the import prefixes to work. So the standard we discussed is `envoyv2` and `contourv2` (for the next PR).

**Pull request merged**